src/Swarm/TUI/Controller.hs

```diff
@@ -719,7 +719,46 @@ resetREPL t r ui =
 
 -- | Handle a user input event for the REPL.
 handleREPLEvent :: BrickEvent Name AppEvent -> EventM Name AppState ()
-handleREPLEvent = \case
+handleREPLEvent x = do
+  s <- get
+  let controlMode = s ^. uiState . uiREPL . replControlMode
+  case x of
+    MetaKey 'p' ->
```

Comment:
Slight tweak: let's add an empty REPL check so that you do not accidentally overwrite the command you are writing.

We could also add it to the `drawKeyMenu` so that the players know it is not available while writing.

Just a suggestion though, if you think it is fine, feel free to leave it as is @kostmo. 😉